### ISSUE TITLE
Big Sur Messages App Breaks

### ISSUE THREAD

**Issue by tylerweitzman** [pqrs-org/Karabiner-Elements#2538]
Hi,
When I use Karabiner on Big Sur, no matter what my actual settings in the app are (even if it's empty), my iMessages app exhibits weird behavior.

In particular, the following functionality breaks:
1. When typing in auto complete in a new compose message, arrow keys and return key hot keys do not trigger
2. When typing a message in the input bar, arrow keys do not work.

Other hot keys do work:
- Enter to send works
- Shift + Enter for new line works
- Option + Arrow keys work to move the cursor an entire word.

Analysis
After some experimentation, I've discovered the following:
1. Indeed when I disable input monitoring and close karabiner-elements the arrow keys start working again on Messages
2. When I try to map another key on karabiner to a down_arrow and use that key it doesn't do anything on iMessage
3. When I try to map one of the arrow keys to a spacebar and then use it on IMessage it does indeed write a space

It appears the issue then is that Karabiner is unable to properly output the arrow key character and the return key into iMessage. SO while it can observe the input of an arrow key it is not able to relay to iMessage. Is it possible to solve this by somehow forcing Karabiner to not intercept arrow keys on iMessage?

Or is there some way to get Messages to accept the input?

Thanks!!!

**Comment by elkind00** [comment]
I'm surprised there are no other responses here.  I encountered the keydef problem with Messages immediately upon installing Big Sur.  I use both DefaultKeyBinding.dict and Karabiner on my Mac, but for Messages I only use DefaultKeyBinding.dict.  As soon as I discovered the problems (several ignored key definitions) I went to Karabiner and defined one of them there (just to make sure it would work), only to discover that it ignored the Karabiner keydef , too.

It'd be really great if this could be looked in to.  Thanks!
